selectinf/randomized/tests/test_posterior.py

```diff
@@ -160,9 +165,11 @@ def test_flexible_prior1(nsample=100, nburnin=50):
     M = E.copy()
```

Comment:
The fact that this test fails seems to mean that you are modifying the state of the query when you run the sampling. This is because when we rerun with identical prior and initial seed we get different results. We shouldn't modify the state if we can help it.

Reply:
This seems fixed now.